packages/flutter/lib/src/rendering/error.dart

```diff
@@ -50,6 +51,9 @@ class RenderErrorBox extends RenderBox {
 
   /// The message to attempt to display at paint time.
   final String message;
+  final double systemTopPadding = Zone.current['systemTopPadding'] == null
```

Comment:
I mean anywhere that you would currently type `ui.window` and the analyzer would know it was a Window, the autocomplete would know it had a `padding` member, the VM would verify the type, etc. Setting side and getting side. You lose all of that with zones.

We have more than one static in dart:ui. For example, you could use zones to mock all the methods, all the getters and setters for callbacks, all the functions, etc. Suppose you wanted to test that a particular hashCode function called hashValues, for example. You could by putting hashValues into a Zone.current map and requiring that it be accessed that way. Similarly if we put the constructors for Path and company into Zone.current, we could provide mocks for those classes.

I think if we wanted to go down a route like that we should prefer to ask for explicit support from Dart for mocking a library. It would presumably be reasonably straight-forward for the VM team to provide a way to substitute another library in any time we `import 'dart:ui'`. This would let us inject an entire mock library, which could re-export from the "real" dart:ui the parts that it doesn't want to mock, providing only the specific parts it wants to override.

Anyway I don't want to beat a dead horse so I'll shut up now. :-)